rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/LocalDataTransferServer.java

```diff
@@ -176,8 +176,11 @@ public void run()
                     throw new Exception("Could not create BookKeeper Client " + Throwables.getStackTraceAsString(e));
                 }
 
-                if (!bookKeeperClient.readData(remotePath, offset, readLength, header.getFileSize(), header.getLastModified(), header.getClusterType())) {
-                    throw new Exception("Could not cache data required by non-local node");
+                if (!CacheConfig.isParallelWarmupEnabled(conf)) {
```

Comment:
if isParallelWarmupEnabled is true, then arent we going ahead with reading the local file without ensuring that the data is present or not?

Reply:
nvm, looks like we we never get here if isParallelWarmupEnabled is true.